### ISSUE TITLE
Basic Tests ObjC MacOS failure

### ISSUE THREAD

**Issue by yashykt** [grpc/grpc#20686]
```
Test Case '-[CallAPIv2Tests testTimeout]' started.
E1018 15:41:58.156512000 123145329954816 ssl_transport_security.cc:1270] Handshake failed with fatal error SSL_ERROR_SSL: error:100000f7:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER.
E1018 15:41:58.157067000 123145329954816 ssl_transport_security.cc:1270] Handshake failed with fatal error SSL_ERROR_SSL: error:100000f7:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER.
/Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_opt_native/src/objective-c/tests/UnitTests/APIv2Tests.m:412: error: -[CallAPIv2Tests testTimeout] : ((error.code) equal to (GRPCErrorCodeDeadlineExceeded)) failed: ("14") is not equal to ("4")
Test Case '-[CallAPIv2Tests testTimeout]' failed (0.006 seconds).
Test Case '-[CallAPIv2Tests testTimeoutBackoff1]' started.
Test Case '-[CallAPIv2Tests testTimeoutBackoff1]' passed (0.002 seconds).
Test Case '-[CallAPIv2Tests testTimeoutBackoff2]' started.
Test Case '-[CallAPIv2Tests testTimeoutBackoff2]' passed (0.002 seconds).
Test Case '-[CallAPIv2Tests testUserAgentPrefix]' started.
Test Case '-[CallAPIv2Tests testUserAgentPrefix]' passed (0.007 seconds).
Test Suite 'CallAPIv2Tests' failed at 2019-10-18 15:41:58.172.
	 Executed 15 tests, with 1 failure (0 unexpected) in 4.643 (4.651) seconds
```

**Comment by yashykt** [comment]
Another instance with different test sets - https://source.cloud.google.com/results/invocations/61e764c7-6310-445f-8e18-5cb1a71ddfe6/targets/github%2Fgrpc%2Frun_tests%2Fobjc_macos_opt_native%2Fmac-test-basictests/tests

**Comment by muxi** [comment]
Fixed by #22327
